### PR TITLE
Switching to update helper module

### PR DESCRIPTION
**Pull request by mtodor**
Make sure these boxes are checked before submitting your pull request - thank you!

- [ ] All coding styles are fulfilled. ([How to check for cs issues?](https://github.com/BurdaMagazinOrg/thunder-dev-tools/blob/master/README.md#code-style-guidelines))
- [ ] All tests are running locally. ([How to run the test?](https://github.com/BurdaMagazinOrg/thunder-distribution/blob/develop/docs/development.md#how-to-run-the-tests))
- [ ] Necessary update hooks are provided.
- [ ] User roles have correct access for new introduced permission.
- [ ] Every thunder module has a README.md in its root. Follow [this guidelines](https://www.drupal.org/node/2181737), but we don't need every topic.
- [ ] Code is covered with well-balanced amount of inline comments.

If you are really awesome, then your feature is covered by additional tests. Well done!

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -143,6 +143,7 @@
         "drupal/slick_media": "^1.0",
         "drupal/thunder_admin": "^1.0 || ^2.0-beta15",
         "drupal/token": "^1.0",
+        "drupal/update_helper": "dev-1.x",
```
Comment (dbosen) [inline composer.json]: we should not forget to release this before merging the PR

File: modules/thunder_article/thunder_article.post_update.php [hunk 2]
```diff
@@ -22,14 +24,25 @@ function thunder_article_post_update_move_scheduler_to_content_list() {
     if ($view) {
       $view->set('id', 'locked_content_old');
       $view->save();
-      $thunderUpdater->logger()->alert('The locked content view was renamed to views.view.locked_content_old.');
+      $updater->logger()->alert('The locked content view was renamed to views.view.locked_content_old.');
```
Comment (dbosen) [inline modules/thunder_article/thunder_article.post_update.php]: Use the $updateLogger

File: modules/thunder_article/thunder_article.post_update.php [hunk 3]
```diff
@@ -22,14 +24,25 @@ function thunder_article_post_update_move_scheduler_to_content_list() {
...
     }
     if ($config_update->import('view', 'locked_content')) {
-      $thunderUpdater->logger()->info('The new locked content view was imported as views.view.locked_content.');
-      $thunderUpdater->checklist()->markUpdatesSuccessful(['thunder_article_move_scheduler_to_content_list']);
+      $updater->logger()->info('The new locked content view was imported as views.view.locked_content.');
```
Comment (dbosen) [inline modules/thunder_article/thunder_article.post_update.php]: $updateLogger again

File: modules/thunder_liveblog/thunder_liveblog.install [hunk 4]
```diff
@@ -102,30 +102,31 @@ function thunder_liveblog_update_8101() {
...
+  if ($updater->executeUpdate('thunder_liveblog', 'thunder_liveblog__update_8102')) {
     $updateLogger->info(t('New form display successfully installed.'));
   }
   else {
     $updateLogger->warning(t('There was an error during the update of the liveblog form display.'));
   }
 
   // Output logged messages to related channel of update execution.
-  return $thunderUpdater->logger()->output();
+  return $updater->logger()->output();
```
Comment (dbosen) [inline modules/thunder_liveblog/thunder_liveblog.install]: and the $updateLogger again

File: modules/thunder_liveblog/thunder_liveblog.install [hunk 5]
```diff
@@ -102,30 +102,31 @@ function thunder_liveblog_update_8101() {
...
 }
 
 /**
  * Switch to new experimental paragraphs widget and paragraphs features.
  */
 function thunder_liveblog_update_8103() {
-  /** @var \Drupal\thunder_updater\Updater $thunderUpdater */
-  $thunderUpdater = \Drupal::service('thunder_updater');
+  /** @var \Drupal\update_helper\Updater $updater */
+  $updater = \Drupal::service('update_helper.updater');
```
Comment (dbosen) [inline modules/thunder_liveblog/thunder_liveblog.install:125]: now we completely miss out on the $updateLogger? consistency my ass! :)

File: modules/thunder_paragraphs/thunder_paragraphs.install [hunk 6]
```diff
@@ -106,60 +106,21 @@ function thunder_paragraphs_update_8001() {
...
-    $updateLogger->info(t('Configuration of media paragraph has been successfully updated.'));
-    $thunderUpdater->checklist()->markUpdatesSuccessful(['v1_1__media_paragraph'], FALSE);
-  }
-  else {
-    $updateLogger->warning(t('Unable to update configuration for media paragraph.'));
-    $thunderUpdater->checklist()->markUpdatesFailed(['v1_1__media_paragraph']);
-  }
-
-  return $updateLogger->output();
+  return $updater->logger()->output();
```
Comment (dbosen) [inline modules/thunder_paragraphs/thunder_paragraphs.install:113]: OK, i stop obsessing about the $updateLogger now...

File: updates_checklist.yml [hunk 7]
```diff
@@ -280,3 +280,14 @@ v2.34:
         '#title': 'Hide content lock messages on other forms than node edit.'
         '#description': '<p>Configuration is successfully updated.</p>'
         '#description_failed': '<p><strong>Update of configuration has failed.</strong><p>'
+v2.35:
+    '#title': 'Update to Thunder 2.33'
```
Comment (dbosen) [inline updates_checklist.yml]: There is a version mismatch

File: modules/thunder_updater/thunder_updater.install [hunk 8]
```diff
@@ -5,23 +5,105 @@
...
+  /** @var \Drupal\Core\Config\ConfigFactoryInterface $config_factory */
+  $config_factory = \Drupal::configFactory();
+
+  // We let exception bubble in this case, since it's not possible to continue
+  // with execution of update hooks without "update_helper.updater" service.
+  $module_installer->install(['update_helper']);
+
+  /** @var \Drupal\update_helper\Updater $updater */
+  $updater = \Drupal::service('update_helper.updater');
+  $update_logger = $updater->logger();
```
Comment (dbosen) [inline modules/thunder_updater/thunder_updater.install:30]: I could start obsessing again now, you know? We now have $update_logger $updateLogger and $update->logger() in use :-)
Btw. I like $update->logger() the best.

File: modules/thunder_updater/thunder_updater.install [hunk 9]
```diff
@@ -5,23 +5,105 @@
...
+
+  // We need database connection in order to migrate Thunder updates correctly.
+  $database = \Drupal::database();
+
+  // Existing module states should be preserved!
+  $has_thunder_updater_checklist = $module_handler->moduleExists('checklistapi');
```
Comment (dbosen) [inline modules/thunder_updater/thunder_updater.install]: Should we not check for the existence of "checklistapi.progress.thunder_updater" config as well? Could have been removed by site builder.

File: modules/thunder_updater/thunder_updater.install [hunk 10]
```diff
@@ -5,23 +5,105 @@
...
+      $update_checklist = \Drupal::service('update_helper_checklist.update_checklist');
+      $update_checklist->markAllUpdates(FALSE);
+
+      $update_checklist->markUpdatesSuccessful(['thunder' => ['thunder_updater_update_8001']]);
+    }
+    catch (Exception $e) {
+      $update_logger->warning('Unable to install "update_helper_checklist".');
+    }
+
+    // We do not need entities created on "update_helper_checklist" install.
```
Comment (dbosen) [inline modules/thunder_updater/thunder_updater.install]: Why is that? What kind of entities are on that list?

**Review by dbosen — CHANGES_REQUESTED** [review]

File: modules/thunder_updater/thunder_updater.install [hunk 11]
```diff
@@ -5,23 +5,105 @@
...
+      }
+      catch (EntityStorageException $e) {
+        $update_logger->warning(sprintf('Unable to delete Update entity (%s) for Update Helper Checklist.', $update->id()));
+
+        $update_checklist->markUpdatesFailed(['thunder' => ['thunder_updater_update_8001']]);
+      }
+    }
+
+    $table_name = $database->getFullQualifiedTableName('thunder_updater_update');
+    $thunder_updater_updates = $database->query("SELECT * FROM {$table_name};")->fetchAll(PDO::FETCH_ASSOC);
```
Comment (dbosen) [inline modules/thunder_updater/thunder_updater.install]: why all the hassle? The {thunder_updater_update} in the query takes care of the correct table name

**Review by dbosen — CHANGES_REQUESTED** [review]

File: modules/thunder_updater/thunder_updater.install (hunk 8 above)
Comment (mtodor) [reply modules/thunder_updater/thunder_updater.install:30]: `$update->logger()` is used, only if we use Logger once, fe. when we need output.
`$updateLogger` or `$update_logger` are used when we use Logger more than once, fe. when we want to add logs inside update hook. Difference between snake case and camel case depends on other variable names in a file (or hook). 😭

File: modules/thunder_updater/thunder_updater.install (hunk 11 above)
Comment (mtodor) [reply modules/thunder_updater/thunder_updater.install]: 👍

File: modules/thunder_updater/thunder_updater.install (hunk 10 above)
Comment (mtodor) [reply modules/thunder_updater/thunder_updater.install]: When `update_helper_checklist` module is installed, all checklist entries are marked as checked and that action creates an `Update` entity for every checklist entry. That's why we have to remove them and then migrate the current status.

I've changed comment a bit in order to be more understandable.

File: modules/thunder_updater/thunder_updater.install (hunk 9 above)
Comment (mtodor) [reply modules/thunder_updater/thunder_updater.install]: I have added a check for this.

**Review by dbosen — APPROVED** [review]

**Pull request merged**